\chapter{Introduction}
\label{chpt:introduction}

The development of Next Generation Sequencing (NGS) has revolutionized the study of genetic variation, allowing for the analysis of many genomes in a fast and cost-efficient manner \citep{Bagger2024}. These developments have enabled the study of the genetic components of a vast number of heritable traits \citep{Sollis2022, Szustakowski2021}, and such results are being used to both identify and characterize biological mechanisms underlying diseases and traits \citep{Gallagher2018}, and to develop polygenic risk scores to better predict an individual's risk of developing certain diseases and better personalize medical care \citep{Lewis2020}. In addition, the generation of large numbers of genome sequences has also allowed for the study of genome diversity and evolution through the analysis of genome-wide patterns of selection, recombination and linkage disequilibrium, and mutation processes \citep{Marchi2021}.

In order to best make use of the information from these studies, it is critical to model the processes underlying the generation and transmission of genetic variation in populations. Foundational to many population genetic models are mutation rates, which are known to vary across the genome \citep{Carlson2018, Aggarwala2016, Zhu2017}. The units in which genetic material is transmitted from parent to child are chromosomes, and as such demographic models attempt to model or simulate the transmission of chromosomes forwards or backwards through time \citep{Hanchard2006, Marchi2021, Sabeti2002}. When we sequence individuals using short-read NGS, we observe the genotypes at positions along the genome, but we do not know the phase of the variants, meaning we do not know which chromosome the variants travel on relative to each other. In the case of samples of unrelated individuals, we rely on statistical phasing algorithms to infer the phased chromosomes. Modern statistical phasing algorithms \citep{Browning2021, Hofmeister2023, Loh2016} are able to generate highly accurate phasing results for large samples in a computationally efficient manner, but even with their high accuracy these algorithms generate thousands of switch errors per genome. 

In chapter II of this dissertation, we provide a deep exploration of the influence of local nucleotide sequence context on the rate of germline mutation at individual positions in the genome. Singleton mutations from the 1000 Genomes Project phase III unrelated samples are used as a proxy for new variants, and we evaluate the distributions of nucleotides at positions flanking these variants to characterize germline mutation processes active in modern human populations.  We employ a control-matching approach in order to control for other genomic features that influence the rate of mutation, allowing for a direct evaluation of the influence of neighboring nucleotides on rates of mutation.

In chapter III of this dissertation, we perform a direct evaluation of three modern statistical phasing algorithms (Eagle v2.4.1, Beagle 5.4, and SHAPEIT 5.1.0). By constructing synthetic diploid chromosomes with known phase by sampling male X chromosomes, we are able to directly evaluate phasing quality over large numbers of samples. In addition to benchmarking the performance of the three methods, we characterize the overlap of errors across methods and correlate the distributions of  errors across the chromosome with genomic features such as recombination rates and GC content. We also rephrase the proband from 602 trios and compare the results to pedigree-adjusted phase in order to evaluate switch error rates along the autosomes.

In chapter IV of this dissertation, we explore model-based approaches to combining the results of multiple statistical phasing algorithms to improve phasing accuracy. Given the large overlap of errors across methods we observe in chapter III, we find that a consensus voting approach leads only to a modest improvement in phasing accuracy. Our aims in this chapter are to model probabilities of errors for each method, identify genomic features that might suggest when we might favor one methodâ€™s inferred phase when there is not a unanimous consensus, and ultimately construct a model to refine phase estimates using multiple estimates. We will also explore using such an approach to refine multiple phase estimates from the same statistical phasing algorithm run on different reference panels.