Mutation rates the key feature of many demographic inference tools, and as the source of all heritable genetic variation, the rates at which they occur along the genome along with the evolution of the underlying processes are themselves the focus of much investigation \citep{Carlson2018, Chintalapati2020, Gao2019, Harris2017, Moore2021, Narasimhan2017, Seplyarskiy2023, Seplyarskiy2021b, Seplyarskiy2021a, Sgurel2014}. In evaluations of mutation rate models, various genomic features have been identified as being associated with either increased or decreased rates of mutations; such features include early/late replication timing, recombination rates, methylation markers, and the nucleotide sequence context of the putative mutation site \citep{Carlson2018, Seplyarskiy2023}. In particular, \citep{Carlson2018} found that the influence of genomic features such as DNAse hypersensitive sites and GC content were associated with both increased and decreased rates of mutation, depending on both the type of mutation under consideration, and the sequence context surrounding the putative mutation site. When incorporating local sequence context in models of mutation rates, choices must be made in regards to how many neighboring positions are to be included, and to what extent interactions among neighboring nucleotides also are to be accounted for. A straightforward-way to incorporate local sequence context into mutation rate models is to consider mutation subtypes not only by the reference and alternative allele at a focal position, but also by the nucleotides at surrounding positions. By fitting separate models for each motif-mutation category, the influence of local sequence context is controlled for not only the individual effects of each included flanking position, but all orders of interaction among the nucleotides at those positions. Taking such an approach, \citep{Carlson2018} found that models which included flanking nucleotides in a $\pm 3$ window surrounding the focal position outperformed models that incorporated fewer flanking positions. However, the addition of additional flanking positions in the motif-category approach causes an exponential increase in the number of categories, and thus limits the size of local sequence context windows used in practice. \citep{Seplyarskiy2023} employed a more flexible approach by augmenting a $\pm 2$ base-pair motif-category model with independent effects for neighboring bases further up/down-stream from the focal position. This assumes that the influence of interactions among neighboring positions on rates of mutation at a focal position are contained within the $\pm 2$ base-pair window, and that the additional influence of additional neighboring positions is captured through independent parameters with no additional interactions. While it seems reasonable to assume that neighboring positions closer a focal position would exert more influence on the probability of a mutation, these assumptions have yet to have been fully explored, and it is an open question as to how far from a focal position would we expect neighboring nucleotides to have an influence on mutation rates, and to what extent and degree do interactions among neighboring sequences have influence on mutation rates.
 
In the generation of NGS data, what we observe are the genotypes at positions along the genome. However, in many applications we would like to know not only which combinations of variants an individual carries along their genomes, but also which variants traveled with each other on the same chromosomes. This requires the reconstruction of phased chromosomes, which can be performed in a number of ways. For short-read NGS data, we generally rely on two approaches: applying the laws of Mendelian inheritance to construct haplotypes when related pedigrees of individuals are sequenced, or statistical phasing methods when we do not have samples from related pedigrees. Modern statistical phasing algorithms \citep{Browning2021, Hofmeister2023, Loh2016} are able to generate highly accurate phasing results for large samples in a computationally efficient manner, but their accuracy depends largely on sample size and reference composition, and errors are generated.

\section{Mutate Rate Extra Text}
A challenge in understanding the influence of local sequence context on mutation rates is in disentangling the effect of neighboring nucleotides on rates of mutation from that of other genomic features. In their evaluation of nucleotide motifs and germline mutation, \citep{Zhu2017} proposed a matched-control approach in which each variant is matched to a nearby non-variant site whose nucleotide in the reference genome matched the reference allele of the variant. In doing so, by controlling for genomic location one would also control for genomic features also associated with genomic location, yielding a set of control non variant sites affected by the same genomic features as their matched singletons. Comparisons of the distributions of nucleotides flanking genomic variants could be compared to that of their matched controls in order to identify motifs associated with mutation. In their work, using common non-coding variants from ENSBML along with location-matched controls, \citep{Zhu2017} identified putative mutation motifs associated with mutation rates for several mutation subtypes. However, these results are limited in several respects, most notably in that they are based on patterns of nucleotides flanking common, non-coding variants sampled from ENSEMBL, whose distributions may have also been shaped by forces such as natural selection and biased gene conversion. Analyses were also primarily contained within the +/- 4 base window centered on the focal site, with evaluation of the marginal influence of individual positions only considered at positions up to 10 bases from the variant site. It remains unclear to what extent nucleotides at further flanking positions influence rates of mutation, and to what extent higher-order interactions among neighboring nucleotides moderate each other’s effects.

Recent studies have indicated potential differences in mutation processes among different ancestry groups \citep{Harris2017, Mathieson2017}. \citep{Harris2017} identified enrichment of several 3-mer mutation categories in cross-population comparisons, suggesting that underlying mutation processes may differ between populations. In their examination of a particular category (TCC -> TTC), they found that the minor allele frequency distribution of the variants underlying the observed enrichment had a pulse at moderate frequencies, suggesting that not only had there been a difference between the underlying mutagenic processes, but that mutation rates could have evolved quickly over short evolutionary timescales. Furthermore, \citep{Aikens2019} found that for several polymorphism types, additional genetic context contributed to variability between populations. While these studies suggest that population-level differences in mutation processes may have differed in the past, the question of whether there exist differences in these processes in modern human populations is under-explored.

Across all subtypes the strongest influence is seen within the +/- 4 base pair window centered at the putative mutation site. Individual nucleotide influence typically decreases as positions further from the site of interest are considered, and similarly the strongest two-way interactions are observed at pairs of flanking positions both closer to one another and closer to the focal position. In each mutation subtype we find that statistically significant marginal influences of individual positions extend well beyond the +/- 10 base window considered in previous analyses. Two-way interactions between bases at all pairs of positions within the +/- 10 base window are also statistically significant. The strongest influence for most mutation subtypes is found at individual positions, with CpG to GpG being the only subtype whose strongest influence is found at a pair of positions. The patterns of influence of individual and pairs of flanking positions is found to be highly consistent between the 1kGP super-populations, but shifts in the distributions of nucleotides at influential locations suggest a potential for subtle differences in mutation processes active in modern human populations. Taken together, our results suggest that the marginal influence of individual flanking nucleotides on rates of substitution expand beyond the +/- 1 base pair window considered in mutation signature profiling, and that accounting for larger windows of nucleotide context could be done in manner which does not necessarily require accounting for all orders of interactions among the included nucleotides. 

\subsection{Discussion}

Local sequence context is known to influence mutation rates, but the range in which neighboring positions are influential and to what extent interactions among neighboring nucleotides moderate each other’s influence. Here we thoroughly characterize the marginal influence of individual nucleotides and pairs of nucleotides on germline mutation rates, while also exploring the relative influence of higher order interactions. We expand on previous work by utilizing the distribution of singleton variants to characterize motifs associated with mutation processes while also expanding the scope of the local sequence context under examination. Our results show that, in general, the marginal influence of single position effects are generally the strongest, but both second and third-order interactions among flanking nucleotides are also influential. The influence of first order effects are found to be strongest at sites nearer to the focal site of interest, but the decrease in the influence as further positions are considered is not monotonic for any mutation subtype.  Comparing patterns of local sequence context influence across distinct ancestral groups reveals patterns of both shared and distinct influence of local sequence context, suggesting that germline mutation processes active in modern human populations are similar but may exhibit some population specific behavior. Altogether, our results offer new insights on the role of local sequence context on rates of mutation and how it should be considered in modeling germline mutation rates.

Previous efforts to either characterize mutation motifs associated with variation in the germline mutation rate or incorporate local sequence context in mutation rate models utilized the distributions of common, non-coding variants and their local sequence contexts \citep{Aggarwala2016, Zhu2017}. However, the distribution of these variants is also influenced by other forces, such as natural selection and biased gene conversion.  Singleton variants are younger relative to common variants and thus their distributions offer a less distorted view of the characteristics of the underlying germline mutation processes \citep{Carlson2018}. Examples of mutation rate models incorporated other genomic and epigenomic features as well as local sequence context, but such efforts have not focused specifically on characterizing the marginal influence of local sequence context \citep{Aggarwala2016, Carlson2018}. However, these approaches rely on having an available measure of the genomic and epigenomic features which influence rates of mutation, and it is unclear whether all such features have been identified, nor is it clear whether available genome annotations are accurate or shared across populations. To disentangle the influence of local sequence context from that of other genomic features, \citep{Zhu2017} compared distributions of nucleotides flanking non-coding variants to corresponding distributions of nucleotides flanking nearby non-variant sites that are expected to be affected by the same genomic and epigenomic features as the variant. Here, we utilize the same matched non-variant control site approach using singleton variants from a higher-quality dataset consisting of samples from multiple, diverse populations. With its high coverage, the 1kGP deep-sequencing dataset, the more than 30 million singleton variants across five distinct super-populations, which allows for both the investigation of the influence of nucleotides at flanking positions and the identification of population-level differences in the influence of local sequence context. 

Neighboring nucleotides are also known to influence rates of somatic mutation. Mutation signatures, inferred from collections of somatic mutations across various cancer types and tissues and characterized by their mutation subtype and adjacent nucleotides within the +/-1 window centered on the mutation, describe both known and putative mutagenic processes \citep{Alexandrov2020}. Each of these signatures reflects not only the types of substitutions that are characteristic of a process or a set of processes, but also the contexts in which the mutations occur, as defined by the immediately flanking nucleotides. The 3-mer categories are symmetric in the sense that the mutation appears in the middle of the motif, and such symmetric categories have also previously been utilized in the study of germline mutation as well \citep{Aggarwala2016, Carlson2018, Harris2017}. While the symmetric 3-mer motif accounts for both the influence of the immediate flanking position and the interaction between the two positions, this may not necessarily capture the most influential two-way interaction among flanking positions. Restricting our analysis to 3-mer motifs, we identify a setting where a non-symmetric 3-mer motif categorization outperforms the symmetric 3-mer in the task of discriminating between singletons and matched controls from a test dataset using rates inferred from the 1kGP data. In this case, the mutation subtype of interest (A>G) had its most influential two-way effect at the (-2, -1) positions; other subtypes have their highest signal at the (-1, +1) pair of positions, in which case the symmetric 3-mer characterization would be preferable. Exploring non-symmetric motif definitions may provide a pathway to adaptable models that bypass the problem of a rapid rise in category numbers caused by the inclusion of additional neighbors to symmetric motifs.

Previous studies of the 1kGP populations have revealed differences in mutation spectra amongst these groups, suggesting that the mutation rate has experienced swift evolutionary progress within a brief evolutionary period, and that populations carry their own variants in the genes responsible DNA replication and repair \citep{Harris2017}. Here we utilize the distributions of singletons, as opposed to common non-coding variants, to assess the influence of local sequence context on rates of mutation. Since singletons are newer in expectation on an evolutionary time scale, their distributions likely reflect mutational processes active in modern human populations. We find that the influence of nucleotides at flanking positions and pairs of positions are consistent across the super-populations, but the results of cross-population models suggests that differences in the distributions of nucleotides at positions flanking mutations exist between populations at influential positions. Thus, while the patterns of influence of local sequence context influence on rates of substitution are consistent across populations, there may be subtle differences between populations regarding the influence of certain arrangements of nucleotides at the positions and interactions shown to be most influential. 

Here, we present the first thorough investigation of the influence of local sequence context on rates of mutation using a large, diverse sample of unrelated individuals sequenced at a high depth. While our results suggest that influence extends beyond windows generally considered in models of the mutation rate, we also find diminishing returns as higher orders of interactions among neighboring nucleotides is considered, suggesting that larger windows of local sequence context could be incorporated into models of mutation in a manner that does not account for all possible orders of interaction among constituent nucleotides. The approach based around categorizing mutations by the symmetric motif centered at the variant site has been a widely used approach to account for the influence of local sequence context, and it has successfully been used to infer characteristics of the biological processes underlying somatic and germline mutation \citep{Alexandrov2020}. Even so, many inferred signatures have an unknown etiology. Future work may be able to further refine our understanding of the biological processes underlying mutations by considering local sequence context in a framework that does not rely on the k-mer categorization approach, but rather on more flexible models which include larger windows of neighboring nucleotides and a more selective inclusion of interactions between flanking nucleotides.

\subsection{Incorporating my results into models of germline mutation rates}

My results suggest a few ways in which models can more flexibly account for local sequence context in modeling germline mutation rates:

\begin{itemize}
  \item Influence of individual flanking positions extend well beyond the $\pm 1$ to $\pm 4$ positions considered in prior approaches \citep{Aggarwala2016, Carlson2018, Seplyarskiy2023} 
  \item The influence of interactions decreases with additional orders added
  \item The most influential effects are those from flanking positions closest to the focal position.
\end{itemize}

These results suggest that a more flexible modeling approach which accounts for the influence of more flanking positions while ignoring unneccessary orders of interactions could lead to better models of the mutation rate along the genome. However, a challenge here would be to develop a method which could quickly explore the space of possible models quickly and yield robust models that generalize to datasets other than the training data used to fit the models.

\subsection{Deep Learning Approaches}

Recent work has utilized large language models to build mutation rate predictions for the human genome (among others) \citep{Fang2022}. Such models have the potential to capture complex relationships between the sequence context around a focal position and the rate of mutation at that position. What I am interested in and where I see interactions between my work and these models are:

\begin{itemize}
    \item Do these sequence-based deep learning models perform better than prior methods based on 7-mer sequence contexts and genomic features?
    \item Based on a trained model, can we extract meaningful measures of flanking position influence on mutation rates and the influence of different orders of interaction on rates of mutation? For example, from a trained model, can we identify signatures of putative biological mechanisms underlying germline mutation rate variaton along the genome?
\end{itemize}

\section{Phasing}

Modern sequencing studies have greatly increased the availability of genetic information from individuals across a wide range of ancestral backgrounds, providing a rich resource for a wide variety of analyses. Whole-genome and whole-exome sequencing (WGS/WES) studies of large cohorts has allowed for the identification of rare variants and their contributions to disease risk and trait heritability (\citep{Wainschtein2022, Wang2021}). Analyses beyond identifying variant-trait associations often require phased haplotypes, which are not directly observed in WGS or WES studies. For example, the identification of compound heterozygous events, wherein both copies of a gene contain different heterozygous variants, requires accurate phase information. Other downstream analyses, such as imputation(\citep{Howie2012,Das2016,Das2018}), demographic inference (\citep{Maples2013,Baran2012,SalterTownshend2019}), and testing for natural selection (\citep{Browning2020NS, Sabeti2002, Hanchard2006, Zhang2006}) either require or prefer phased haplotypes. Population-based phasing methods allow for the statistical inference of haplotypes in large samples of unrelated individuals. Many of these methods implement the model of \cite{Li2003} along with computational techniques to allow for the phasing of biobank-scale sample sizes across the entire human genome \cite{Browning2021, Hofmeister2023, Loh2016}. While the availability of larger samples and high quality reference panels have led to improvements in phasing accuracy, modern phasing algorithms still introduce thousands of switch errors in each phased genome \cite{Choi2018}. These errors impact downstream analyses, and therefore a comprehensive analysis of their frequency and the genomic contexts in which they occur more frequently is of considerable interest.

Phasing errors occur when variants at heterozygous sites in a diplotype are placed on the wrong chromosome relative to variants at other heterozygous sites. The measure commonly used to measure phasing quality is the switch rate, which is the proportion of pairs of consecutive heterozygous sites which are not correctly phased. If a single switch occurs within a genomic region, then haplotypes around the error are incorrectly inferred for the individual. However, if switches occur at two consecutive heterozygous positions, then haplotypes in a region will have been correctly inferred except for the single heterozygous position at which the first switch occurred \citep{Browning2022}. We distinguish between the former and latter as switches and flips, respectively.

Previous efforts to benchmark phasing methods have relied on the availability of either trio genotyping or sequencing data, or genomes phased using a combination of multiple sequencing platforms and experimental techniques \citep{Choi2018}. While the pedigree information allows for the construction of phased haplotypes under the principles of Mendelian inheritance, switch and flip error rates estimated from trio data have been shown to be inflated due to genotyping error \citep{Browning2022}. In addition, the cost of obtaining parent-child trios often precludes their widespread use in genetic studies, whereas large samples of unrelated individuals are readily available \citep{ByrskaBishop2022}. A phasing evaluation which leveraged such a large sample would allow for a more detailed characterization of phasing errors across chromosomes. However, the absence of gold-standard haplotypes in this setting does not allow for a direct evaluation of phasing methods. Thus previous efforts to benchmark phasing methods using these large samples of unrelated individuals have relied on imputation quality as a downstream metric of phasing quality \citep{Stahl2021, DeMarino2022}. Imputation accuracy is a reflection of the underlying phasing quality and thus allows for a comparison across methods, but it does not allow for a detailed characterization of the genomic context in which phasing errors occur as the number and location of phasing errors are unknown. 

A direct evaluation of statistical phasing methods on samples of unrelated individuals that identifies the quantity and location of phasing errors requires true haplotypes to serve as a benchmark. While haplotype phase is initially unknown for most sequenced chromosomes, the phase of male X chromosomes is known outside of the pseudo-autosomal region (PAR). This suggests a strategy of constructing “synthetic diploids” with known phase by sampling two male X chromosomes. These synthetic diploids can then be phased and compared to the known phase to both estimate rates of phasing errors and characterize the locations where errors occur. While limited to the X chromosome, such an approach allows for the evaluation of phasing in studies without requiring the presence of trios. Large studies of unrelated individuals across diverse populations are readily available, and synthetic diploids offer an avenue by which the rate of statistical phasing errors can be evaluated and characterized with regards to the genomic contexts in which they occur.

We construct synthetic X chromosome diploids by sampling male X chromosomes from unrelated samples in the 1kGP 30x high-coverage whole genomes data set. Each synthetic diploid consists of two male X chromosomes sampled from the 2,504 phase 3 1kGP subjects. 200 synthetic diploids are generated for each of the five 1kGP super populations for a total of 1000 synthetic diploids. Each synthetic diploid is then individually phased with three modern statistical phasing methods (Beagle 5.4, Eagle 2.4, SHAPEIT5), utilizing a reference panel consisting of the 2,502 1kGP phase 3 samples who were not used to construct the synthetic diploid. The reconstructed haplotypes are then compared to the original X chromosomes to identify phasing errors.

In our benchmark of the three methods, we find that Beagle (198 errors/sd) and SHAPEIT(201 errors/sd) introduce fewer errors on average than Eagle (237errors/sd).  When considering switches and flips separately, we observe that Eagle (146.34 switches/sd) and Beagle (107.367 switches/sd) tend to generate more non-flip switches than SHAPEIT (98.313 switches/sd), while for flips we observe that SHAPEIT (102.89 flips / sd) introduces more flip errors on average than either Beagle (90.367 flips/sd) or Eagle(90.187 flips/sd).

\subsection{Methods}

\begin{itemize}
    \item Add equations
\end{itemize}

We utilize male X chromosomes to construct synthetic diploids of known phase in order to evaluate three model statistical phasing algorithms \citep{Browning2021, Hofmeister2023, Loh2016}. The chromosomes are sampled from the 2,504 phase 3 samples of the 1000 Genomes Project (1kGP), allowing for comparisons to be made across distinct ancestral groups. In addition to calculating error rates, we also characterize the genomic context in which errors occur by comparing error densities across the chromosome to other genomic features. Additionally we re-phase chromosome 15 for children from 602 trio pedigrees without pedigree-based correction and evaluate the discordance between the population-based phasing inferred haplotypes and the pedigree-adjusted phase.

The high-coverage WGS expanded 1kGP cohort consists of the 2,504 original phase 3 1kGP samples, along with an additional 602 phased trios. These samples all are members of one of five super-populations (AFR, AMR, EAS, EUR, SAS), with each super-population consisting of non-overlapping subsets of 26 distinct populations \citep{ByrskaBishop2022}. For the construction of the synthetic chromosome X diploids we use only the 2,504 phase 3 unrelated individuals. From the phased VCF for chromosome X we remove sites that are in either pseudo-autosomal region (PAR1 and PAR2), and we also remove sites that are difficult to sequence according to the 1kGP pilot accessibility mask \citep{auton2015}. From the remaining variant sites we select only biallelic SNPs.  

For our analysis of the 602 1kGP trios, we filter the phased VCF for chromosome 15 to retain only SNPs that are biallelic, are not singletons, and are not in masked regions of the chromosome in the 1kGP pilot accessibility mask. The unphased VCF is filtered to the same set of variants, removing any SNPs which are multiallelic in the unphased VCF that are biallelic in the phased VCF. 

Each synthetic diploid consists of two male X haploids sampled from two individual samples from the same population. For each super-population, we first sample a population from which we sample two individual male X chromosomes. With the sampled haplotypes, we create a phased VCF file with the known phase and an unphased VCF where at heterozygous sites the order of the two alleles is selected at random. Each synthetic diploid has its own reference panel constructed from the 2,502 samples not used in the synthetic diploid. We construct 200 synthetic diploids for each of the five 1kGP super-populations. Each synthetic diploid is phased with Eagle v2.4.1, Beagle 5.4, and SHAPEIT5 using the reference panel crafted for each pseudodiploid. For genetic maps, we utilize the maps provided by software’s authors \cite{Browning2021,Hofmeister2023,Loh2016}.

For our synthetic X diploids, we identify the location of switch and flip errors by comparing the inferred diploids from each of the three phasing methods to the original haploids used in construction of the synthetic diploid. Using Vcftools (version 0.1.17), we identify the location of all discordances between the inferred phase and the known true phase. Each discordance is classified as either a switch or a flip, where flips are defined as sets of two consecutive errors at consecutive heterozygous positions, while any remaining error is classified as a switch. We follow the convention described in \citep{Browning2022} and consider each flip to be one error in total even though they appear as two errors in the output from vcftools.

\subsubsection{1kGP Trio Proband Re-phasing}
For each of the 602 1kGP trios, we phase chromosome 15 of each child without pedigree-based correction. For the reference panel we use the phased 1kGP chromosome 15 vcf which we subset to the 2,504 1kGP phase 3 unrelated samples. For each child, we further subset this panel to remove the parents or the child if either were part of the phase 3 sample. We phase each of the 602 children using the same methods as we used for the synthetic X chromosome diploids, and compare the population-based inferred phase to the pedigree-based phase provided in the 1kGP phased VCF.